### PR TITLE
[Fix] Fix optimizer parameter override bugs.

### PR DESCRIPTION
**Pull request by Baibaifan**
fix(training): apply scheduler overrides before restoring optimizer lr/wd.

Restore checkpoint weight-decay scheduler fields before reapplying the scheduler step during load_state_dict. This ensures weight_decay written to optimizer param groups uses checkpoint scheduler values when override is not enabled.

Also restore runtime param-group scheduler fields in override mode so optimizer checkpoint state cannot override max_lr/min_lr/start_wd/end_wd.

`--override-opt-param-scheduler`

Tests:
- python -m tests/unit_tests/test_optimizer_param_scheduler.py

### CODE REVIEW

**Review by FDecaYed — APPROVED** [review]

**Comment by Baibaifan** [comment]
> /ok to test [82b8f07](https://github.com/NVIDIA/Megatron-LM/commit/82b8f07ff63c1e0412d308d19e6577fcb9a45ac6)

hi @FDecaYed 

I adjusted the DCO signature; could you please help me reactivate the unit test checks?

**Comment by Baibaifan** [comment]
> /ok to test [432e4fc](https://github.com/NVIDIA/Megatron-LM/commit/432e4fc606640ef63b0fd6f8ffaa7fc25629faf3)

The unit tests have all passed, please merge. @yaox12 @FDecaYed

**Comment by yaox12** [comment]
It should be added to the merge train automatically when CI passes. Not sure why it didn't work. I just retried it.
